### PR TITLE
Adds simple framework for learnable crafting recipes

### PR DESCRIPTION
**Pull request by AnturK**
To restrict what can be handcrafted by whom.
Also removes a bit of copypaste from granter books.

### CODE REVIEW

**Comment by Borisvanmemes** [comment]
I can already see this being used for a new traitor item that teaches you how to macgyver fuckloads of silly and hardly working weapons, and i love it.

**Comment by Barhandar** [comment]
I can see it being used to prevent assistants from crafting anything and I hate it.

**Comment by anconfuzedrock** [comment]
inb4 lizards know a cooking recipe for popcorn chicken popsicles

**Comment by MrPerson** [comment]
When it comes to actually hooking up the mob with the thing being taught, I love how datum actions and martial arts call a function on the DA/MA itself with the mob as an arg, spells call a function on the mob's mind with a spell instance as an arg, and crafting recipes call a function on the mob itself with the recipe type as an arg. It's so consistent!

**Comment by AnturK** [comment]
Spells have both variations because we have both mind-bound and mob-bound spells. I suppose i could normalize action granting, there's not really any specific reason it's called that way but that's kinda out of scope of this.

**Comment by PKPenguin321** [comment]
job restricted innate abilities a shit

**Comment by MrPerson** [comment]
Yeah I know it's way out of scope, it's just sad :frowning_face:

**Comment by knacker48** [comment]
If your going to add it can you make so whats available now is available to everyone, and have new things down the line such as specific upgrades/traitor items/ some way of obtaining more recipes by locked?

File: code/modules/crafting/craft.dm [hunk 1]
```diff
@@ -431,3 +435,19 @@
...
 
 	return data
+
+//Mob helpers
+
+/mob/proc/known_recipes(datum/crafting_recipe/R)
+	if(!mind)
+		return FALSE
+	if(!mind.learned_recipes)
+		return FALSE
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/crafting/craft.dm]: can't this just be rewritten as 
```DM
if(mind?.learned_recipes)
    return (R.type in mind.learned_recipes)
return FALSE
```

Comment (SpaceManiac) [reply code/modules/crafting/craft.dm]: if you are golfing this, `return R.type in mind?.learned_recipes` because `_ in null` is always false

Comment (AnturK) [reply code/modules/crafting/craft.dm]: Opted with moving these to mind to keep this consistent with spells.

**Pull request merged**